nordicsemi/dfu/dfu_transport_serial.py

```diff
@@ -445,31 +448,32 @@ def __get_checksum_response(self):
 
     def __stream_data(self, data, crc=0, offset=0):
         logger.debug("Serial: Streaming Data: " +
-            "len:{0} offset:{1} crc:0x{2:08X}".format(len(data), offset, crc))
+                     "len:{0} offset:{1} crc:0x{2:08X}".format(len(data), offset, crc))
+
         def validate_crc():
             if (crc != response['crc']):
-                raise ValidationException('Failed CRC validation.\n'\
-                                + 'Expected: {} Recieved: {}.'.format(crc, response['crc']))
+                raise ValidationException('Failed CRC validation.\n'
+                                          + 'Expected: {} Recieved: {}.'.format(crc, response['crc']))
             if (offset != response['offset']):
-                raise ValidationException('Failed offset validation.\n'\
-                                + 'Expected: {} Recieved: {}.'.format(offset, response['offset']))
+                raise ValidationException('Failed offset validation.\n'
+                                          + 'Expected: {} Recieved: {}.'.format(offset, response['offset']))
 
-        current_pnr     = 0
+        current_pnr = 0
 
-        for i in range(0, len(data), (self.mtu-1)/2 - 1):
+        for i in range(0, len(data), int((self.mtu-1)/2 - 1)):
```

Comment:
For some weird reason, `self.mtu-1` was uneven in our tests, even though we could not explain this behavior. This is the reason, why `int(...)` was added all over the place during debugging. Your suggested way of using the integer division is way better, but we need to be careful that self.mtu is not float, since otherwise, a TypeError exception will occur.

E.g. `1.0 // 2 = 0.0 ; 1 // 2 = 0`